internal/handlers/pg/pgdb/indexes.go

```diff
@@ -22,25 +22,61 @@ import (
 	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 )
 
-// indexParams describes the parameters for creating an index.
+// indexParams contains parameters for creating an index.
+// TODO This type will become exported in https://github.com/FerretDB/FerretDB/issues/1509 (similar to QueryParams).
 type indexParams struct {
-	schema   string // pg schema name
-	table    string // pg table name
-	isUnique bool   // whether the index is unique
+	db         string   // FerretDB database name
+	collection string   // FerretDB collection name
+	index      string   // FerretDB index name
+	key        indexKey // Index specification (pairs of field names and sort orders) // TODO
+	unique     bool     // Whether the index is unique
+}
+
+// indexKey defines a type for index key - pairs of field names and sort orders.
+type indexKey []indexKeyPair
+
+// indexKeyPair consists of a field name and a sort order that are part of the index.
+type indexKeyPair struct {
+	field string
+	order indexOrder
+}
+
+// indexOrder defines a type for index sort order.
+type indexOrder int8
+
+// indexOrder constants.
+const (
+	indexOrderAsc  indexOrder = 1
+	indexOrderDesc indexOrder = -1
+)
+
+// createIndex creates a new index for the given params.
+// TODO This method will become exported in https://github.com/FerretDB/FerretDB/issues/1509.
+func createIndex(ctx context.Context, tx pgx.Tx, ip *indexParams) error {
+	pgTable, pgIndex, err := newMetadata(tx, ip.db, ip.collection).setIndex(ctx, ip.index, ip.key, ip.unique)
+	if err != nil {
+		return err
+	}
+
+	if err := createIndexIfNotExists(ctx, tx, ip.db, pgTable, pgIndex, true); err != nil {
+		return err
+	}
+
+	return nil
 }
 
 // createIndexIfNotExists creates a new index for the given params if it does not exist.
-func createIndexIfNotExists(ctx context.Context, tx pgx.Tx, p *indexParams) error {
+func createIndexIfNotExists(ctx context.Context, tx pgx.Tx, schema, table, index string, isUnique bool) error {
 	var err error
 
 	unique := ""
-	if p.isUnique {
+	if isUnique {
 		unique = " UNIQUE"
 	}
 
-	sql := `CREATE` + unique + ` INDEX IF NOT EXISTS ` + pgx.Identifier{p.table + "_id"}.Sanitize() +
-		` ON ` + pgx.Identifier{p.schema, p.table}.Sanitize() +
-		` ((_jsonb->>'_id'))`
+	sql := `CREATE` + unique + ` INDEX IF NOT EXISTS ` + pgx.Identifier{index}.Sanitize() +
+		` ON ` + pgx.Identifier{schema, table}.Sanitize() +
+		` ((_jsonb->'_id'))` // TODO Provide ability to set fields https://github.com/FerretDB/FerretDB/issues/1509
```

Comment:
Sorry for reviewing the same line second time but we could try to simplify it to one pair of brackets, or even no brackets

```suggestion
		` _jsonb->'_id'` // TODO Provide ability to set fields https://github.com/FerretDB/FerretDB/issues/1509
```

Reply:
@noisersup did you try to run it locally? I tried your suggestion but it failed as syntax error. So I think double brackets are needed.

Reply:
What about single brackets?

Reply:
syntax error.

Reply:
` ERROR: syntax error at or near "->" (SQLSTATE 42601)`